lightwood/analysis/nc/nc.py

```diff
@@ -337,6 +338,17 @@ def score(self, x, y=None):
 
         return self.err_func.apply(prediction, y) / norm
 
+    def __deepcopy__(self, memo={}):
+        cls = self.__class__
+        result = cls.__new__(cls)
+        memo[id(self)] = result
+        for k, v in self.__dict__.items():
+            if k != 'model':  # model should not be copied
+                setattr(result, k, deepcopy(v, memo))
+            else:
+                setattr(result, k, v)
+        return result
```

Comment:
will steal this for another PR since I think it could be a great timesaver, also <i think> the normalizer should also be ignored here.